components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/ext/APIManagerExtensionHandler.java

```diff
@@ -71,9 +75,19 @@ public boolean mediate(MessageContext messageContext, String direction) {
     public boolean handleRequest(MessageContext messageContext) {
         Timer.Context context = startMetricTimer(DIRECTION_IN);
         long executionStartTime = System.nanoTime();
+        TracingSpan requestMediationSpan = null;
+        if (Util.tracingEnabled()) {
+            TracingSpan responseLatencySpan =
+                    (TracingSpan) messageContext.getProperty(APIMgtGatewayConstants.RESPONSE_LATENCY);
+            TracingTracer tracer = Util.getGlobalTracer();
+            requestMediationSpan = Util.startSpan(APIMgtGatewayConstants.REQUEST_MEDIATION, responseLatencySpan, tracer);
+        }
         try {
             return mediate(messageContext, DIRECTION_IN);
         } finally {
+            if (Util.tracingEnabled()) {
```

Comment:
Add catch block and handle the error cases.

Reply:
Fixed